### PR TITLE
Add attribute to store verification pending mobile number

### PR DESCRIPTION
**Pull request by DinikaSen**
### Proposed changes in this pull request

Adds a new attribute to store the newly updated mobile number until it is verified by the user.

Related to issue - https://github.com/wso2/product-is/issues/9045

### CODE REVIEW

**Comment by dewniMW** [comment]
@IsuraD, @emswbandara do we have to introduce a ldap schema for identity claims as we can change the underlying persistence of identity claims to a userstore based storage?

File: features/org.wso2.carbon.ldap.server.server.feature/resources/conf/identityPerson.ldif [hunk 1]
```diff
@@ -160,6 +166,6 @@ objectClasses: ( 1.3.6.1.4.1.37505.1.101
     SUP scimPerson
     STRUCTURAL
     MAY  ( primaryChallenges $ challenges $ firstChallenge $ secondChallenge $ oneTimePassword $ accountLock $ temporaryEmail $ recoveryEmail $ passwordChangeRequired $ passwordTimestamp $ temporaryLock $ lastFailedAttemptTime $ failedLoginAttempts $ lastLogonTime $ unlockTime $ verifyEmail $ askPassword $ forcePasswordReset $ failedRecoveryAttempts $ primaryChallengeQuestion $ emailVerified $ challengeQuestionUris $ 
-    $ lastLoginTime $ lastPasswordUpdate $ phoneVerified $ accountDisabled $ preferredChannel $ lockedReason $ pendingEmailAddress)
+    $ lastLoginTime $ lastPasswordUpdate $ phoneVerified $ accountDisabled $ preferredChannel $ lockedReason $ pendingMobileNumber )
```
Comment (IsuraD) [inline features/org.wso2.carbon.ldap.server.server.feature/resources/conf/identityPerson.ldif]: Why are we removing pendingEmailAddress?

Comment (DinikaSen) [reply features/org.wso2.carbon.ldap.server.server.feature/resources/conf/identityPerson.ldif]: Addressed in https://github.com/wso2-extensions/identity-userstore-ldap/pull/39/commits/f586cf881757961d0c0e00e9af6e3495e09c1a05

**Comment by DinikaSen** [comment]
PR builder passed : https://jenkins-support-all.wso2.com/jenkins/job/DEV_PR_BUILDER/854/

**Review by ashensw — APPROVED** [review]

**Pull request merged**